core/src/androidMain/kotlin/BluetoothDeviceAndroidPeripheral.kt

```diff
@@ -159,19 +177,19 @@ internal class BluetoothDeviceAndroidPeripheral(
     }
 
     override suspend fun disconnect() {
-        connectAction.resetAndJoin()
-        setDisconnected()
+        _connection?.bluetoothGatt?.disconnect()
+        suspendUntil<Disconnected>()
         logger.info { message = "Disconnected" }
     }
 
-    private fun close() {
-        connectAction.reset()
-        setDisconnected()
```

Comment:
This was unnecessary, as the state `Flow` is updated via `Callback`.

https://github.com/JuulLabs/kable/blob/63731b40f774d7d6b0715a6d487f5f21f9270422/core/src/androidMain/kotlin/gatt/Callback.kt#L106-L111

Reply:
Correction: we still need to explicitly set disconnected state if we only call `close` (and don't call `disconnect`) — which we do in some situations (e.g. BLE gets disabled or peripheral scope is disposed w/o first disconnecting peripheral).